### PR TITLE
Allow Safari versions above 13 to play HVEC

### PR DESCRIPTION
**Pull request by Southpaw1496**
**Changes**
Currently, the client is hardcoded not to allow HVEC playback on Safari even though it supports it, because "some version was lying about support". Since modern versions of Safari do support HVEC, I've added a condition to check if the Safari version is greater than or equal to Version 13, since prior versions required macOS 10.13 High Sierra to support HVEC, and so I assume that is where Safari "lying" about support became a problem. Ideally, I think the hardcoding should be removed altogether, but this is a gentler solution.

**Issues**
- #4216

### CODE REVIEW

File: src/scripts/browserDeviceProfile.js [hunk 1]
```diff
@@ -537,7 +537,7 @@ export function canPlaySecondaryAudio(videoTestElement) {
 
         if (canPlayHevc(videoTestElement, options)) {
             // safari is lying on HDR and 60fps videos, use fMP4 instead
-            if (!browser.safari) {
+            if (!browser.safari || (browser.safari && browser.version >= 13)) {
```
[the remark this answers is not in the document]
Comment (Southpaw1496) [reply src/scripts/browserDeviceProfile.js:540]: When I was testing (as I want to have my library in HVEC eventually), Safari wouldn't play any H265 video without transcoding, even if I ticked "Prefer fMP4 HLS" in Advanced Playback settings, and dmitrylyzo said that that code was the culprit in #4216

[the remark this answers is not in the document]
Comment (nyanmisaka) [reply src/scripts/browserDeviceProfile.js:540]: Can you share the media info of your HEVC video?

And debug these two fields to see why fMP4 is not available in your Safari:

https://github.com/jellyfin/jellyfin-web/blob/c5e0b66d6afc42bf459adf6e410b9e995b26f4d8/src/scripts/browserDeviceProfile.js#L10-L25

https://github.com/jellyfin/jellyfin-web/blob/c5e0b66d6afc42bf459adf6e410b9e995b26f4d8/src/scripts/browserDeviceProfile.js#L715-L727

[the remark this answers is not in the document]
Comment (Southpaw1496) [reply src/scripts/browserDeviceProfile.js:540]: Here are the properties of the video I'm using for testing (it's a Bugs Bunny short)

Screenshot 2023-01-30 at 9 50 10 am

I don't know what you mean by debugging the fields, I'm still quite new to web development.

[the remark this answers is not in the document]
Comment (nyanmisaka) [reply src/scripts/browserDeviceProfile.js:540]: > User policy for root. EnablePlaybackRemuxing: False EnableVideoPlaybackTranscoding: False EnableAudioPlaybackTranscoding: False

I see this from your last post. Can you enable remuxing in the user settings. fMP4 playback requires remuxing(copy from mkv or mp4 to fmp4).

[the remark this answers is not in the document]
Comment (Southpaw1496) [reply src/scripts/browserDeviceProfile.js:540]: I enabled "Allow Media Playback that doesn't require transcoding" and the file now plays with a black screen. If I enable all the transcoding settings, it plays correctly, but with the type "Transcoding"

SCR-20230130-ed7

Screenshot 2023-01-30 at 10 26 13 am

[the remark this answers is not in the document]
Comment (nyanmisaka) [reply src/scripts/browserDeviceProfile.js:540]: Can you see "HEVC (direct)" in transcoding info with only the "Allow Media Playback that doesn't require transcoding" enabled?

If yes then please share a ffmpeg-remux* log.

[the remark this answers is not in the document]
Comment (Southpaw1496) [reply src/scripts/browserDeviceProfile.js:540]: How would I access the log?

I am running Jellyfin inside of Docker.

[the remark this answers is not in the document]
Comment (nyanmisaka) [reply src/scripts/browserDeviceProfile.js:540]: Jellyfin -> Dashboard -> Log -> FFmpeg.Transcode*.log

[the remark this answers is not in the document]
Comment (Southpaw1496) [reply src/scripts/browserDeviceProfile.js:540]:
```
/videos/31aaa79d-3f2b-063a-23f9-d17cfd3016fc/hls1/main/71.ts

{"Protocol":0,"Id":"31aaa79d3f2b063a23f9d17cfd3016fc","Path":"/media/Movies/8 Ball Bunny (1950).mp4","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mov,mp4,m4a,3gp,3g2,mj2","Size":113741773,"Name":"8 Ball Bunny (1950)","IsRemote":false,"ETag":"fdfbdd9cdc993d86abf770a978545d7b","RunTimeTicks":4278000000,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"hevc","CodecTag":"hvc1","Language":"und","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"DvVersionMajor":null,"DvVersionMinor":null,"DvProfile":null,"DvLevel":null,"RpuPresentFlag":null,"ElPresentFlag":null,"BlPresentFlag":null,"DvBlSignalCompatibilityId":null,"Comment":null,"TimeBase":"1/12800","CodecTimeBase":null,"Title":null,"VideoRange":"SDR","VideoRangeType":"SDR","VideoDoViTitle":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"576p HEVC SDR","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":1871451,"BitDepth":8,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":576,"Width":720,"AverageFrameRate":25,"RealFrameRate":25,"Profile":"Main","Type":1,"AspectRatio":"4:3","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":90,"IsAnamorphic":null},{"Codec":"aac","CodecTag":"mp4a","Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"DvVersionMajor":null,"DvVersionMinor":null,"DvProfile":null,"DvLevel":null,"RpuPresentFlag":null,"ElPresentFlag":null,"BlPresentFlag":null,"DvBlSignalCompatibilityId":null,"Comment":null,"TimeBase":"1/48000","CodecTimeBase":null,"Title":null,"VideoRange":null,"VideoRangeType":null,"VideoDoViTitle":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"Eng - AAC - Stereo - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"stereo","BitRate":249341,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":2,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":"LC","Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":2127008,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -analyzeduration 200M -ss 00:07:06.000 -fflags +genpts -f mov,mp4,m4a,3gp,3g2,mj2  -i file:"/media/Movies/8 Ball Bunny (1950).mp4" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -tag:v:0 hvc1 -bsf:v hevc_mp4toannexb -start_at_zero -codec:a:0 copy -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type mpegts -start_number 71 -hls_segment_filename "/config/transcodes/1287c2417829b4007c105a9da6ebf3fc%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/transcodes/1287c2417829b4007c105a9da6ebf3fc.m3u8"


ffmpeg version 5.1.2-Jellyfin Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-libs=-lfftw3f --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-chromaprint --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --toolchain=hardened --enable-cross-compile --arch=arm64 --cross-prefix=/usr/bin/aarch64-linux-gnu-
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'file:/media/Movies/8 Ball Bunny (1950).mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    creation_time   : 2022-12-18T15:08:13.000000Z
    encoder         : Lavf59.34.102
  Duration: 00:07:07.80, start: 0.000000, bitrate: 2127 kb/s
  Stream #0:0[0x1](und): Video: hevc (Main) (hvc1 / 0x31637668), yuv420p(tv, progressive), 720x576 [SAR 16:15 DAR 4:3], 1871 kb/s, 25 fps, 25 tbr, 12800 tbn (default)
    Metadata:
      creation_time   : 2022-12-18T15:08:13.000000Z
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
      encoder         : Lavc59.54.100 libx265
  Stream #0:1[0x2](eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 249 kb/s (default)
    Metadata:
      creation_time   : 2022-12-18T15:08:13.000000Z
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
Output #0, hls, to '/config/transcodes/1287c2417829b4007c105a9da6ebf3fc.m3u8':
  Metadata:
    encoder         : Lavf59.27.100
  Stream #0:0: Video: hevc (Main) (hvc1 / 0x31637668), yuv420p(tv, progressive), 720x576 [SAR 16:15 DAR 4:3], q=2-31, 1871 kb/s, 25 fps, 25 tbr, 90k tbn (default)
  Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 249 kb/s (default)
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
frame=    0 fps=0.0 q=-1.0 size=N/A time=00:00:00.00 bitrate=N/A speed=N/A    
[hls @ 0xaaaae997b4d0] Opening '/config/transcodes/1287c2417829b4007c105a9da6ebf3fc71.ts' for writing
frame=  207 fps=0.0 q=-1.0 Lsize=N/A time=00:00:08.32 bitrate=N/A speed= 539x    
video:1024kB audio:263kB subtitle:0kB other streams:0kB global headers:2kB muxing overhead: unknown
```

**Comment by thornbill** [comment]
I think something went wrong with your merge/rebase...

>  These merge commits were added into this branch cleanly.

**Comment by Southpaw1496** [comment]
It's always possible, I couldn't get the dev environment to work to test that it was successful.

**Comment by Southpaw1496** [comment]
But I definitely did have to resolve a conflict or two in order to merge.

**Comment by Southpaw1496** [comment]
All fixed :)

**Comment by bleak183** [comment]
I just had a very similar issue where most of my library which is transcoded to AAC wouldnt play on safari. Enabling HEVC was the fix. I wonder why this was never merged???

**Comment by nyanmisaka** [comment]
@Southpaw1496 This is superseded by #5183

**Comment by thornbill** [comment]
👍 I'll close

**Pull request closed without merging**